File: src/main/java/com/matag/cards/ResourceLoader.java
package com.matag.cards;

import org.springframework.core.io.Resource;
import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
import org.springframework.stereotype.Component;

import lombok.SneakyThrows;

@Component
public class ResourceLoader {

  @SneakyThrows
  public Resource[] getCardsFileNames() {
    return new PathMatchingResourcePatternResolver(this.getClass().getClassLoader())
        .getResources("/cards/*");
  }

  @SneakyThrows
  public Resource[] getSetsFileNames() {
    return new PathMatchingResourcePatternResolver(this.getClass().getClassLoader())
        .getResources("/sets/*");
  }
}

File: src/main/kotlin/com/matag/cards/ResourceLoader.kt
package com.matag.cards

import org.springframework.core.io.support.PathMatchingResourcePatternResolver
import org.springframework.stereotype.Component

@Component
class ResourceLoader {
    fun getCardsFileNames() = PathMatchingResourcePatternResolver(this.javaClass.classLoader)
            .getResources("/cards/*")

    fun getSetsFileNames() = PathMatchingResourcePatternResolver(this.javaClass.classLoader)
            .getResources("/sets/*")
}
